### PR TITLE
feat: add event and interval annotation support to chart data ep

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Implements event, interval and line annotations on V1 chart data endpoint and ECharts Timeseries plugin. Also refactor the annotation unit test fixtures by moving them into `tests/annotation_layers/fixtures.py` so they can more easily be used in other tests.

### SCREENSHOT
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by john-bodley** [comment]
@junlincc this still looks like WIP. @villebro could you ping this thread when the PR is ready for review?

**Comment by villebro** [comment]
@john-bodley sure thing, I'll add some tests, will ping you shortly once it's done.

File: tests/annotation_layers/api_tests.py [hunk 1]
```diff
@@ -29,77 +27,17 @@
...
-                )
-            yield annotation_layers
-
-            # rollback changes
-            for annotation_layer in annotation_layers:
-                db.session.delete(annotation_layer)
-            for annotation in annotations:
-                db.session.delete(annotation)
-            db.session.commit()
-
```
Comment (villebro) [inline tests/annotation_layers/api_tests.py:102]: These fixtures were moved to a separate `fixtures.py` file to make them more reusable in different contexts (in this case for testing the chart data endpoint)

**Comment by villebro** [comment]
@john-bodley this is now ready for review if you have time, sorry for the delay

File: tests/charts/api_tests.py [hunk 2]
```diff
@@ -1383,3 +1386,44 @@ def test_import_chart_invalid(self):
         assert response == {
             "message": {"metadata.yaml": {"type": ["Must be equal to Slice."]}}
         }
+
+    @pytest.mark.usefixtures("create_annotation_layers")
```
Comment (dpgaspar) [inline tests/charts/api_tests.py:1390]: We should think about moving all these tests away from `unittest.TestCase` if possible, there are some really cool pytest features that we are missing out

File: superset/common/query_context.py [hunk 3]
```diff
@@ -220,7 +222,80 @@ def cache_key(self, query_obj: QueryObject, **kwargs: Any) -> Optional[str]:
...
+            result = {"columns": columns, "records": records}
+            annotation_data[layer_name] = result
+        return annotation_data
+
+    @staticmethod
+    def get_viz_annotation_data(
+        annotation_layer: Dict[str, Any], force: bool
+    ) -> Dict[str, Any]:
+        slice_id = annotation_layer["value"]
+        slc = db.session.query(Slice).filter_by(id=slice_id).one_or_none()
```
Comment (dpgaspar) [inline superset/common/query_context.py:264]: nit: could make sense to use `ChartDao.find_by_id` instead, note DAO's are starting to support being called from celery workers and the Security chart filter is always applied there

**Review by dpgaspar — APPROVED** [review]
LGTM, just of couple of comments

File: tests/annotation_layers/fixtures.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (bkyryliuk) [inline tests/annotation_layers/fixtures.py:1]: this looks great, more common way to make fixtures available is to use conftest:
https://www.tutorialspoint.com/pytest/pytest_conftest_py.htm

tests/annotation_layers/conftest.py - will make them automatically available to all files in the modules and submodules

File: tests/charts/api_tests.py (hunk 2 above)
Comment (bkyryliuk) [reply tests/charts/api_tests.py:1390]: +1, some functions like login, get_table_by_name should be already available as direct ( not object ) methods

**Review by bkyryliuk — APPROVED** [review]
Looks really good, just 2 nits / suggestions

File: superset/charts/api.py [hunk 5]
```diff
@@ -485,6 +485,7 @@ def data(self) -> Response:
         """
         if request.is_json:
             json_body = request.json
+            print(json_body)
```
[the remark this answers is not in the document]
Comment (villebro) [reply superset/charts/api.py:488]: 🤦  done

File: superset/common/query_context.py (hunk 3 above)
Comment (villebro) [reply superset/common/query_context.py:264]: Good point, I always forget about these! 👍

File: tests/annotation_layers/fixtures.py (hunk 4 above)
Comment (villebro) [reply tests/annotation_layers/fixtures.py:1]: Thanks, I wasn't aware of this feature! I looked into the effort required for properly implementing this, and I believe it makes sense to do that in a separate follow-up PR, as I think I'll end up shuffling around quite a few files.

**Pull request merged**